src/utils/gcsApi.ts

```diff
@@ -234,4 +236,104 @@ export class CraftGCSClient {
       logger.info(`[dry-run] Skipping upload for \`${filename}\``);
     }
   }
+
+  /**
+   * Downloads a file stored on the artifact provider
+   *
+   * @param artifact The artifact to download
+   * @param destinationDirectory Path to directory into which to download the
+   * file
+   * @returns Path to the downloaded file
+   */
+  public async downloadArtifact(
```

Comment:
Since this is a lower-level util function, it doesn't really have to know about artifacts.
Looks like we only need the remote path and the resulting filename from the artifact, perhaps we can just pass those values directly, what do you think?

Reply:
👍🏻